rad/src/reducers/hash_concatenate.rs

```diff
@@ -0,0 +1,62 @@
+use crate::{
+    error::RadError,
+    hash_functions::RadonHashFunctions,
+    operators::bytes as bytes_operators,
+    reducers::RadonReducers,
+    types::{array::RadonArray, bytes::RadonBytes, integer::RadonInteger, RadonType, RadonTypes},
+};
+use serde_cbor::value::Value;
+
+/// Reducer that concatenate all the RadonTypes and hash
+#[allow(clippy::cast_precision_loss)]
+pub fn hash_concatenate(input: &RadonArray) -> Result<RadonTypes, RadError> {
+    let value = input.value();
+
+    match value.first() {
+        None => Ok(RadonTypes::from(RadonInteger::from(0))),
+        Some(RadonTypes::Bytes(_)) => {
+            let concatenated_bytes =
+                value
+                    .iter()
+                    .try_fold(vec![], |mut bytes, item| match item {
+                        RadonTypes::Bytes(rad_bytes) => {
+                            let new_bytes = rad_bytes.value();
+                            match new_bytes.len() {
+                                32 => bytes.extend_from_slice(&new_bytes),
+                                x if x < 32 => {
+                                    let diff = 32 - x;
+                                    bytes.extend_from_slice(&new_bytes);
+                                    for _i in 0..diff {
+                                        bytes.extend_from_slice(&[0]);
+                                    }
+                                }
+                                x if x > 32 => {
+                                    bytes.extend_from_slice(&new_bytes[..32]);
+                                }
+                                _ => {
+                                    unreachable!()
+                                }
+                            }
+
+                            Ok(bytes)
+                        }
+                        _ => Err(RadError::MismatchingTypes {
+                            method: RadonReducers::HashConcatenate.to_string(),
+                            expected: RadonBytes::radon_type_name(),
+                            found: item.clone().radon_type_name(),
+                        }),
+                    })?;
+
+            let hash_function = [Value::from(RadonHashFunctions::SHA2_256 as u8)];
```

Comment:
To avoid the clippy warning, we could use `u8::from`:

```suggestion
            let hash_function = [Value::from(u8::from(RadonHashFunctions::SHA2_256))];
```

But that requires adding `#[derive(IntoPrimitive)]` to the definition of `pub enum RadonHashFunctions`.

Reply:
Or we could avoid creating this `Value` by calling `hash_functions::hash` instead of `bytes_operators::hash`.